File: src/mcp-server/tools/logs.py
import subprocess
from typing import Optional

async def query_journal(service: Optional[str] = None, lines: int = 50) -> str:
    cmd = ["journalctl", "--no-pager", "-n", str(lines)]
    if service:
        cmd.extend(["-u", service])
        
    try:
        result = subprocess.run(cmd, capture_output=True, text=True)
        return result.stdout
    except Exception as e:
        return f"Error querying journal: {e}"

File: src/lib/mcp/tools/logs.ts
/* SPDX-License-Identifier: LGPL-2.1-or-later */
import cockpit from "cockpit";

export async function queryJournal(service?: string, lines: number = 50): Promise<string> {
    const cmd = ["journalctl", "--no-pager", "-n", lines.toString()];
    if (service) {
        cmd.push("-u", service);
    }

    try {
        return await cockpit.spawn(cmd);
    } catch (e: unknown) {
        return `Error querying journal: ${e instanceof Error ? e.message : String(e)}`;
    }
}
